Bump to Fedora 36

skitt opened the pull request:
Fedora 36 isn't quite released yet, but it provides Go 1.18, and we
need Go 1.17 for current Kubernetes dependencies (see
https://github.com/submariner-io/lighthouse/pull/658 for example).

(Fedora 35 has Go 1.16.)

Depends on #776
Depends on #802 
Depends on #803
Depends on https://github.com/submariner-io/submariner-operator/pull/2016

Signed-off-by: Stephen Kitt <skitt@redhat.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

skitt committed:
Bump to Fedora 36

Fedora 36 isn't quite released yet, but it provides Go 1.18, and we
need Go 1.17 for current Kubernetes dependencies (see
https://github.com/submariner-io/lighthouse/pull/658 for example).

(Fedora 35 has Go 1.16.)

Signed-off-by: Stephen Kitt <skitt@redhat.com>

mkolesnik reviewed (approved):
Generally makes sense though I wonder if it's not better to wait for official F36 release?

skitt commented:
> Generally makes sense though I wonder if it's not better to wait for official F36 release?

The packages we care about are unlikely to change much before the F36 release at this point.

skitt committed:
Mark the source directory as safe

See
https://github.blog/2022-04-12-git-security-vulnerability-announced/
for context. git now refuses to handle repositories which don't belong
to the current user by default; such repositories need to be
explicitly marked as safe, in the global configuration for the current
user.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

dfarrell07 committed:
Merge branch 'devel' into fedora-36

dfarrell07 commented on package/Dockerfile.shipyard-dapper-base:
```diff
@@ -1,4 +1,4 @@
-FROM fedora:35
+FROM fedora:36
```
(tl;dr this works both pre and post-release)
It seems like this tag is maintained at the latest build from the 36 branch, which is currently pre-release but will be updated to 36 once it's released, and will continue to be updated after release (based on 34/35 tags I see) 


https://quay.io/repository/fedora/fedora?tab=tags
